code/game/objects/structures/survey_probe.dm

```diff
@@ -0,0 +1,125 @@
+/obj/structure/survey_probe
+	name = "survey probe"
+	desc = "\
+		All-in-one survey probe, able to provide preliminary analysis of planetary bodies. \
+		It has different devices, samplers, and drill bits, as well as internal processing computers, \
+		to inspect the atmosphere, ground, soil, crust, and many other properties and qualities of planetary bodies. \
+		Commonly used by surveyors, explorers, pioneers, all over the Spur, looking for planets that are actually worth settling or exploiting for resources. \
+		This probe was manufactured by Orion Express, but it is based on on older model designed by Hephaestus Industries.\
+		"
+	desc_info = "\
+		The probe has to be deployed first before it is used. Wrench it to deploy, then click with empty hand to activate.\
+		"
+	icon = 'icons/obj/structure/survey_probe.dmi'
+	icon_state = "surveying_probe"
+	density = TRUE
+	/// If false, probe is not deployed.
+	/// If true, it is deployed and ready to survey.
+	anchored = FALSE
+	/// Timer of the survey process.
+	/// If null, it is not currently surveying.
+	var/timer_id = null
+
+/obj/structure/survey_probe/attackby(obj/item/item, mob/living/user)
+	if(!timer_id)
+		if(item.iswrench())
+			if(!anchored)
+				user.visible_message(
+					SPAN_NOTICE("\The [user] unfastens the locking bolts on \the [src], deploying it."),
+					SPAN_NOTICE("You unfasten the locking bolts on \the [src], and it deploys, lowering its devices and drill bits to the ground. It is ready to survey."),
+					)
+				item.play_tool_sound(user, 30)
+				anchored = TRUE
+				icon_state = "surveying_probe_deployed"
+			else
+				user.visible_message(
+					SPAN_NOTICE("\The [user] fastens the locking bolts on \the [src], stowing it."),
+					SPAN_NOTICE("You fasten the locking bolts \the [src], stowing it. It retracts its devices and drill bits."),
+					)
+				item.play_tool_sound(user, 30)
+				anchored = FALSE
+				icon_state = "surveying_probe"
+
+/obj/structure/survey_probe/attack_hand(mob/user as mob)
+	if(!timer_id)
```

Comment:
Flip this to a guard or combine the two conditions

Reply:
not sure what you mean there

Reply:
https://en.wikipedia.org/wiki/Guard_(computer_science)
https://www.geeksforgeeks.org/guard-clause-in-javascript/

Reply:
done